examples/asset/remote_asset_loader.rs

```diff
@@ -0,0 +1,167 @@
+                if response.status != 200 {
+                    return Err(AssetReaderError::HttpError(response.status));
+                }
+
+                response.bytes
+            }
+            Err(error) => {
+                warn!("Failed to read remote asset: {error}");
+                return Err(AssetReaderError::HttpError(500));
+            }
+        };
+
+        // Try to save the downloaded asset on disk.
+        if let Err(AssetWriterError::Io(error)) = self.writer.write_bytes(path, &body).await {
+            return Err(AssetReaderError::Io(Arc::new(error)));
+        }
+
+        // Try to read again the asset, since its saved on disk now.
+        self.reader.read(path).await
+    }
+}
+
+impl AssetReader for RemoteAssetLoader {
+    fn read<'a>(
+        &'a self,
+        path: &'a Path,
+    ) -> BoxedFuture<'a, Result<Box<Reader<'a>>, AssetReaderError>> {
+        Box::pin(async move {
+            // First check if the default reader will find the asset local. If it doesn't find
+            // then try to read from CDN.
+            match self.reader.read(path).await {
+                Ok(reader) => Ok(reader),
+                Err(error) => match error {
+                    AssetReaderError::NotFound(_) => {
+                        debug!("Asset {path:?} not found on local disk. Downloading from CDN.");
+                        self.read_remote(path).await
+                    }
+                    _ => Err(error),
+                },
+            }
+        })
+    }
+
+    fn read_meta<'a>(
+        &'a self,
+        path: &'a Path,
+    ) -> BoxedFuture<'a, Result<Box<Reader<'a>>, AssetReaderError>> {
+        // Just forward the call to default reader.
+        self.reader.read_meta(path)
+    }
+
+    fn read_directory<'a>(
+        &'a self,
+        path: &'a Path,
+    ) -> BoxedFuture<'a, Result<Box<PathStream>, AssetReaderError>> {
+        // Just forward the call to default reader.
+        self.reader.read_directory(path)
+    }
+
+    fn is_directory<'a>(
+        &'a self,
+        path: &'a Path,
+    ) -> BoxedFuture<'a, Result<bool, AssetReaderError>> {
+        // Just forward the call to default reader.
+        self.reader.is_directory(path)
+    }
+}
+
+/// A plugins that registers our new asset loader
+struct RemoteAssetLoaderPlugin;
+
+impl Plugin for RemoteAssetLoaderPlugin {
+    fn build(&self, app: &mut App) {
+        app.register_asset_source(
+            AssetSourceId::Default,
+            AssetSource::build().with_reader(|| Box::new(RemoteAssetLoader::new())),
+        );
+    }
+}
+
+fn main() {
+    let mut app = App::new();
+    app.add_plugins((RemoteAssetLoaderPlugin, DefaultPlugins))
+        .add_systems(Startup, setup)
+        .run();
+
+    // Just a cleanup to remove the downloaded font, so the example will be able to download
+    // next time.
+    let _ = std::fs::remove_file("assets/fonts/ProtestRevolution-Regular.ttf");
```

Comment:
do we want it to be cleaned up? keeping it would show that it's used if already present

Reply:
If we would like to no cleanup at least adding to `.gitignore` would be needed, to avoid updating the project files whenever the example is ran.

Reply:
It probably would be best if the example runs identically every time. Maybe a better option is to load a second font using one of the other fonts already in the assets folder (and not remove it after running of course).

Reply:
You mean before running the `App`, "manually" remove the font and then download from bevy repo? One problem is that if the example fails for some reason (like no internet connection), the project would be broken, that's why I'm really confident in messing with a new asset (and removing it after).

Reply:
Sorry if I wasn't clear. I was thinking that in addition to the text that is spawned with the downloaded font, you could also spawn a second bit of text that uses one of the fonts that is already located in the bevy assets folder. This allows the example to be consistent every run, and shows off that it doesn't need to download the font if it already exists locally (as suggested by mockersf).

Reply:
Indeed that's a good suggestion. I'll add another text with an existing local font.